scipy/stats/_continuous_distns.py

```diff
@@ -3282,6 +3283,308 @@ def _ppf(self, q, k, s):
 mielke = mielke_gen(a=0.0, name='mielke')
 
 
+class kappa4_gen(rv_continuous):
+    """Kappa 4 parameter distribution.
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The probability density function for kappa4 is::
+
+        kappa4.pdf(x, h, k) = (1.0 - k*x)**(1.0/k - 1.0)*
+                              (1.0 - h*(1.0 - k*x)**(1.0/k))**(1.0/h-1)
+
+    if ``h`` and ``k`` are not equal to 0.
+
+    If ``h`` or ``k`` are zero then the pdf can be simplified:
+
+    h = 0 and k != 0::
+
+        kappa4.pdf(x, h, k) = (1.0 - k*x)**(1.0/k - 1.0)*
+                              exp(-(1.0 - k*x)**(1.0/k))
+
+    h != 0 and k = 0::
+
+        kappa4.pdf(x, h, k) = exp(-x)*(1.0 - h*exp(-x))**(1.0/h - 1.0)
+
+    h = 0 and k = 0::
+
+        kappa4.pdf(x, h, k) = exp(-x)*exp(-exp(-x))
+
+    kappa4 takes ``h`` and ``k`` as shape parameters.
+
+    The kappa4 distribution returns other distributions when certain
+    ``h`` and ``k`` values are used.
+
+    +------+-------------+----------------+---------------+
+    | h    | k=0.0       | k=1.0          | -inf<=k<=inf  |
+    +======+=============+================+===============+
+    | -1.0 | Logistic    |                | Generalized   |
+    |      |             |                | Logistic(1)   |
+    +------+-------------+----------------+---------------+
+    |  0.0 | Gumbel      | Reverse        | Generalized   |
+    |      |             | Exponential(2) | Extreme Value |
+    +------+-------------+----------------+---------------+
+    |  1.0 | Exponential | Uniform        | Generalized   |
+    |      |             |                | Pareto        |
+    +------+-------------+----------------+---------------+
+
+    (1) The "Generalized Logisitic" test doesn't pass.  There are at
+        least five generalized logistic distributions.  Four are
+        described here:
+        https://en.wikipedia.org/wiki/Generalized_logistic_distribution
+        The "fifth" one is the one kappa4 should match which isn't
+        implemented in scipy:
+        https://en.wikipedia.org/wiki/Talk:Generalized_logistic_distribution
+        http://www.mathwave.com/help/easyfit/html/analyses/distributions/gen_logistic.html
```

Comment:
The kappa4 with h=-1 and k=0 is the _logistic_ distribution.  The genlogistic(c=1) distribution is the logistic distribution.

```
In [6]: gg = genlogistic(c = 1)

In [7]: ll = logistic()

In [9]: max(gg.pdf(x) - ll.pdf(x))
Out[9]: 0.0

In [10]: kk = kappa4(h=-1, k=0)

In [11]: max(ll.pdf(x) - kk.pdf(x))
Out[11]: 5.5511151231257827e-17

```

For the genextreme and genpareto distributions I used the _k_ value from the kappa4 in the tests...

For example, the genpareto test:
```
def test_cdf_genpareto(self):
        # h = 1 and k != 0 is generalized Pareto
        x = [0.0, 0.1, 0.2, 0.5]
        h = 1.0
        for k in [-1.9, -1.0, -0.5, -0.2, -0.1, 0.1, 0.2, 0.5, 1.0,
                  1.9]:
            vals = stats.kappa4.cdf(x, h, k)
            # shape parameter is opposite what is expected
            vals_comp = stats.genpareto.cdf(x, -k)
            assert_allclose(vals, vals_comp)
```

I was not able to figure out the relationship between kappa4(h=-1, k=?) and genlogistic(c=?).  In the one case when k=0 in kappa4 and c=1 in genlogistic the distributions are equal and special case to the logistic distribution.  Does this mean that "c" should equal "k + 1"?  Nope, doesn't work.

```
In [12]: kk = kappa4(h=-1, k=1)

In [13]: gg = genlogistic(c = 2)

In [14]: max(gg.pdf(x) - kk.pdf(x))
Out[14]: 0.27343257296753037
```

Basically, according to the literature kappa4(h=-1, k=?) should equal some shaped genlogistic distribution for all k.  

The "fifth" genlogistic equation I mention in the text is the kappa4 equation with h=-1.